src/App.tsx

```diff
@@ -1,27 +1,106 @@
-import { render } from '@testing-library/react';
-import React from 'react';
-import './App.css';
-import Contacts from './Contacts';
-import Header from './Header';
-import Sidebar from './Sidebar';
-import Feed from './Feed';
+import React, { useEffect, useState } from "react";
+import "./App.css";
+import Contacts from "./Contacts";
+import Header from "./Header";
+import Sidebar from "./Sidebar";
+import Feed from "./Feed";
+import CreatePostWindow from "./CreatePostWindow";
 
 function App() {
+  let username = "Meir";
+
+
+  
+  const [postData, setPostData] = useState([""]);
+  const addPost = (input: string) => {
+    setPostData([input, ...postData]);
+  };
+
+  const [isNewPostWindowAlive, setWindowState] = useState(false);
+  const toggleWindow = () => {
+    if (isNewPostWindowAlive && isNewPostInProcess) setGlobalTemp(""); //dispach input from createPostWindow to MessageSenderPlaceholder
+    console.log("clicked");
+    setWindowState((isNewPostWindowAlive) => !isNewPostWindowAlive);
+    if (!isNewPostInProcess) setGlobalTemp(`What's on your mind, ${username}?`);
+  };
+  // update: A->D && P->P input
+  // update: A->D && P->I placeholder
+
+  const [isNewPostInProcess, setNewPostInProcess] = useState(false);
+  const newPostInProcess = (b: boolean) => {
+    console.log(isNewPostWindowAlive);
+    // if (!b && !isNewPostWindowAlive) setGlobalTemp(`What's onmindxxx, ${username}?`);
+    // if (b) document.getElementsByClassName("create_post__button")[0].setAttribute('style', 'background-color:#2374E1; color:white; cursor:pointer' );
+    // if (!b) document.getElementsByClassName("create_post__button")[0].removeAttribute('style');
+    setNewPostInProcess(b);
+  };
+
+  const [globalTempNewPost, setGlobalTempNewPost] = useState(
+    `What's on your mind, ${username}?`
+  );
+
+  const setGlobalTemp = async (input: string) => {
+    if (input.replace(/\s/g, "") === "") setGlobalTempNewPost(localTempNewPost);
+    else setGlobalTempNewPost(input);
+  };
+
+  const [localTempNewPost, setLocalTempNewPost] = useState("");
```

Comment:
put all states together in the file, one under the other, to make easy to find them

Reply:
What about the handlers? aren't they suppose to be located with each state, too?

Reply:
Like I said before, I really don't see the use of handlers if you don't do anything on them. You can remove them if the only thing you do is setState